field/binary.go

```diff
@@ -112,13 +113,23 @@ func (f *Binary) SetData(data interface{}) error {
 }
 
 func (f *Binary) MarshalJSON() ([]byte, error) {
-	return json.Marshal(f.Value)
+	str, err := f.String()
```

Comment:
From what I see in `json` package:

```go
// Array and slice values encode as JSON arrays, except that
// []byte encodes as a base64-encoded string, and a nil slice
// encodes as the null JSON value.
```

and calling `f.String` here to get string representation of `Value` and then `encoding.ASCIIToHex.Encode` to get it back looks unnecessary.

Maybe we can just `json.Marshal(f.Value)` and then unmarshal it
```go
var []byte value
err := json.Unmarshal(b, &v)
return f.SetBytes(value)
```

Am I missing something?